src/control/configureIndex.ts

```diff
@@ -4,54 +4,50 @@ import {
   ConfigureIndexRequest,
 } from '../pinecone-generated-ts-fetch/control';
 import { PineconeArgumentError } from '../errors';
-import { buildValidator } from '../validator';
 import type { IndexName } from './types';
-
-import { Type } from '@sinclair/typebox';
-import {
-  ReplicasSchema,
-  PodTypeSchema,
-  IndexNameSchema,
-  DeletionProtectionSchema,
-} from './types';
-
-const ConfigureIndexOptionsSchema = Type.Object(
-  {
-    spec: Type.Optional(
-      Type.Object({
-        pod: Type.Object({
-          replicas: Type.Optional(ReplicasSchema),
-          podType: Type.Optional(PodTypeSchema),
-        }),
-      })
-    ),
-    deletionProtection: Type.Optional(DeletionProtectionSchema),
-  },
-  { additionalProperties: false }
-);
+import { ValidateProperties } from '../utils/validateProperties';
 
 export const configureIndex = (api: ManageIndexesApi) => {
-  const indexNameValidator = buildValidator(
-    'The first argument to configureIndex',
-    IndexNameSchema
-  );
-  const patchRequestValidator = buildValidator(
-    'The second argument to configureIndex',
-    ConfigureIndexOptionsSchema
-  );
-
-  return async (
+  const validator = async (
     indexName: IndexName,
     options: ConfigureIndexRequest
-  ): Promise<IndexModel> => {
-    indexNameValidator(indexName);
-    patchRequestValidator(options);
+  ) => {
+    if (options) {
```

Comment:
Example of new utility function `ValidateProperties` in new file `utils/validateProperties.ts`, which takes in a list of acceptable property names and identifies unacceptable property names, if passed.